### PR TITLE
[triggers] Support `Item` as parameter & Add arg type checks

### LINKED ISSUE #176: Error `java.lang.IllegalArgumentException: Invalid type '{com.oracle.truffle.polyglot.PolyglotMap}' of configuration value!` does not show its location in the .js file
When I create a JS rule as in 
```JavaScript
rules.JSRule({
    triggers: my_array.map(y => triggers.ItemStateChangeTrigger(y)),
    execute: (data) => {
      console.log(data)
   }
))
```
and the elements of my_array are Items (not strings), openHAB 3.3 logs:

> 2022-11-24 09:58:11.273 [ERROR] [ipt.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/etc/openhab/automation/js/zz.js': java.lang.IllegalArgumentException: Invalid type '{com.oracle.truffle.polyglot.PolyglotMap}' of configuration value!

The error disappears once I pass to triggers.ItemStateChangeTrigger() a string.

The error message
2022-11-24 09:58:11.273 [ERROR] [ipt.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/etc/openhab/automation/js/zz.js': java.lang.IllegalArgumentException: Invalid type '{com.oracle.truffle.polyglot.PolyglotMap}' of configuration value!

is very unclear on the exact problem and its location.

Please extend openhab-js to print in such cases the line and column, where the error occurs, and hint that the problem is wrong type passed to `triggers.ItemStateChangeTrigger()`.

### PR DESCRIPTION
**Pull request by florian-h05**
Fixes #176.

## Description

- Improve tests for `typeOfArguments`
- Globally mock `@runtime` and OSGi services (preparation for changes to `triggers.js`)
- [triggers] Support `Item` as argument for appropriate triggers
- [triggers] Add argument type checks
- Remove old type definitions
- Exclude some files from type definitions

## Testing

All changes to `triggers.js` are automatically tested by our unit tests.

### CODE REVIEW

**Comment by florian-h05** [comment]
@sfriedle Can you please review? Especially my changes to the unit tests please.

File: test/@runtime.mock.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    getService: jest.fn(),
+    getAllServiceReferences: jest.fn(),
+    registerService: jest.fn()
+  },
+  lifecycle: {
+    addDisposeHook: jest.fn()
+  }
+}), { virtual: true });
+
+const { bundleContext } = require('@runtime/osgi');
```
Comment (sfriedle) [inline test/@runtime.mock.js:25]: I would try to keep the mocks in the setup files as common and simple as possible and to mock as nearby to the requiring tests as possible. In this particular case (the tests in `triggers.spec.ts`), the `@runtime` module doesn't have to be mocked. The usage of the `Item` type from `items/items.js` pulls in this dependency, which is easy to cut with this mock (in `triggers.spec.ts`):

```js
jest.mock('../items', () => ({ Item: class {} }));
```

With this change, you can skip all the mock implementations in `bundleContext` in this file (beyond this line), the mocking of `OSGIServices` and keep the mock of the `@runtime` module cleaner.

To better explain my thoughts, I've created a pr to your branch: florian-h05/openhab-js#1.

Comment (florian-h05) [reply test/@runtime.mock.js]: @sfriedle  Thanks for your help, I've commited your changes. Could you please re-review (possibly also the other changes to JS files) and then approve if everything is fine?

File: test/typeOfArguments.spec.js [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
-      expect(() => typeOfArguments([5, BigInt(5), true, Symbol('foo'), null, 'string'], expectedArray)).toThrowError(TypeError);
+      expect(() => typeOfArguments([5, BigInt(5), true, Symbol('foo'), null, undefined, 'string'], expectedArray)).toThrowError(TypeError);
+    });
+  });
+
+  describe('supports classname checking', () => {
+    const expectedArray = ['Car', 'Bus'];
+    class Car {}
+    class Bus {}
+    it('does not trow an error if arguments match.', () => {
```
Comment (sfriedle) [inline test/typeOfArguments.spec.js:20]: Typo:
```suggestion
    it('does not throw an error if arguments match.', () => {
```

**Review by sfriedle — APPROVED** [review]
I found a typo. For the rest: LGTM.

**Pull request merged**